### ISSUE TITLE
cfa can't read PP file

### ISSUE THREAD

**Issue by JonathanGregory** [NCAS-CMS/cf-python#165]
cfa is unable to read the PP file attached in [example.zip](https://github.com/NCAS-CMS/cf-python/files/5658447/example.zip). It says

    racc-login-0-1$ cfa jaaac.003003m.example.pp
    cfa ERROR reading /storage/basic/baobab/jonathan/general/jaaac.003003m.example.pp: TODO
    racc-login-0-1$ which cfa
    /home/users/pg902583/jonathan/scripts/cfa

Please could you help? Thanks. Jonathan

**Comment by davidhassell** [comment]
Thanks, Jonathan. Having improved the error message somewhat, I now get:

```
>> cfa -vm jaaac.003003m.example.pp 
cfa ERROR reading jaaac.003003m.example.pp: Incompatible units: <Units: days since 7481-1-1 360_day>, <Units: days since 7481-9-1 gregorian>
>>
```
I don't know what this means, but I am investigating ...

**Comment by JonathanGregory** [comment]
Thanks! That's an improvement. This is a multiannual seasonal mean. Perhaps that is causing the problem.

**Comment by davidhassell** [comment]
OK - I've got to the bottom of it. I now get:
```
>> cfa -vm jaaac.003003m.example.pp 
Field: toa_incoming_shortwave_flux (ncvar%UM_m01s01i207_vn405)
--------------------------------------------------------------
Data            : toa_incoming_shortwave_flux(latitude(37), longitude(48)) W m-2
Cell methods    : time(1): mean within years time(1): mean over years
Dimension coords: time(1) = [7481-07-16 00:00:00] 360_day
                : latitude(37) = [90.0, ..., -90.0] degrees_north
                : longitude(48) = [0.0, ..., 352.5] degrees_east
>> 
```
The problem was an interaction between improved error trapping in `cf.Data` and the recent changes in calendar expectation in `cftime`. The fix will be in v3.8.0, out next week.

**Comment by JonathanGregory** [comment]
Lovely, thanks. Could it be made available sooner via /home/users/pg902583/jonatha?
